oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/core/oidc/OidcClientMetadataClaimAccessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.core.oidc;
+
+import org.springframework.security.oauth2.core.ClaimAccessor;
+
+import java.util.List;
+
+/**
+ * @author Ovidiu Popa
+ * @since 0.1.1
+ */
+public interface OidcClientMetadataClaimAccessor extends ClaimAccessor {
+
+	default List<String> getRedirectUris(){
+		return getClaimAsStringList(OidcClientMetadataClaimNames.REDIRECT_URIS);
+	}
+
+	default List<String> getResponseTypes(){
+		return getClaimAsStringList(OidcClientMetadataClaimNames.RESPONSE_TYPES);
+	}
+
+	default List<String> getGrantTypes(){
+		return getClaimAsStringList(OidcClientMetadataClaimNames.GRANT_TYPES);
+	}
+
+	default String getClientName(){
+		return getClaimAsString(OidcClientMetadataClaimNames.CLIENT_NAME);
+	}
+
+	default String getScope(){
+		return getClaimAsString(OidcClientMetadataClaimNames.SCOPE);
+	}
+
+	default String getTokenEndpointAuthenticationMethod() {
```

Comment:
hi @jgrandja . The reason why I left it as string is because here https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata is not defined as an array.

Reply:
Good catch! I admit I find this odd though. It would be more flexible to allow a client to use more than one auth method on the token endpoint. Either way, we need to comply with the spec so let's leave as-is.